### PR TITLE
AP_NavEKF2/3: do not check the second range finder if RANGEFINDER_MAX_INSTANCES is 1

### PR DESCRIPTION
**Pull request by tatsuy**
EKF always checks two rangefinders even if RANGEFINDER_MAX_INSTANCES is 1.
I added the RANGEFINDER_MAX_INSTANCES check.

### CODE REVIEW

File: libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ void NavEKF2_core::readRangeFinder(void)
         // store samples and sample time into a ring buffer if valid
         // use data from two range finders if available
 
-        for (uint8_t sensorIndex = 0; sensorIndex <= 1; sensorIndex++) {
+        for (uint8_t sensorIndex = 0; sensorIndex <= MIN(1, RANGEFINDER_MAX_INSTANCES - 1); sensorIndex++) {
```
[the remark this answers is not in the document]
Comment (tatsuy) [reply libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp]: fixed

**Comment by tatsuy** [comment]
@peterbarker Thank you for your review.
Would you confirm that this change is what you intended.

**Comment by peterbarker** [comment]
> @peterbarker Thank you for your review. Would you confirm that this change is what you intended.

Yep, exactly, thanks :-)

**Review by rmackay9 — APPROVED** [review]
LGTM

**Comment by rmackay9** [comment]
@priseborough,

This is a mostly mechanical replace but could you confirm you're OK with this?  PeterBarker and I have both reviewed it.

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**